### DISCUSSION TITLE
Hook to customize JWT Payload

### DISCUSSION THREAD

**Discussion by joelbqz** [directus/directus#5406] [Feature Requests]
Hey guys, I've discovered this amazing CMS because I've been looking for a CMS that integrates with my existing Postgres database that I use with  Hasura GraphQL Engine, everything works fine, but I also would love to integrate the existing and robust auth system that Directus offers with Hasura as well, but Hasura requires to add their custom claims in the JWT in order for it to work.

Normally other auth services(example: NextAuth, Auth0) let to add custom claims in the JWT payload. 

Best!

**Comment by rijkvanzanten** [comment]
Sounds good to me! Wanna open a PR for that? The token is generated here: https://github.com/directus/directus/blob/main/api/src/services/authentication.ts#L158 so I imagine we'll have a `auth.jwt.before` hook that allows you to alter that JWT payload 🙂

**Reply by joelbqz** [reply]
Yes, I'm happy to provide a pull request for this, I'm not familiar with the Directus codebase but I'll try in my spare time tonight!

**Reply by joelbqz** [reply]
I opened a pull request for this: https://github.com/directus/directus/pull/7906

Best!! 🙌🏼
